### PR TITLE
Issue #2426: Implement MAD Aggregate

### PR DESCRIPTION
**Pull request by hawkfish**
The Moving Absolute Deviation (or MAD) function is to medians what standard deviation is to average.  This CL implements the simple and moving versions of the aggregate. It also moves the quantile benchmark tests from the slow test suite to the benchmark suite. Testing shows that the moving version achieves a 3.5x performance gain (29.5s vs 8.5s) on 10M rows with 100 element fixed frames.

`MAD` is defined on temporal types as well as numeric types, which required adding a few methods to `Interval` to handle exact intervals more elegantly.

### CODE REVIEW

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Cool stuff. Some comments. Otherwise I would like to see the code coverage, but that is dependent on #2491 being merged.

**Review by Mytherin — APPROVED** [review]
Everything looks good now, thanks!

**Review by Mytherin — COMMENTED** [review]
Thanks for the updates! Looks good to me, could you just have a look at the uncovered lines?

File: src/function/aggregate/holistic/quantile.cpp [hunk 1]
```diff
@@ -105,34 +132,32 @@ static idx_t ReplaceIndex(idx_t *index, const FrameBounds &frame, const FrameBou
...
 	D_ASSERT(index);
 
-	auto curr = fdata[index[j]];
+	// NULLs can replace NULLs
+	// and NULLs are at the end, so we only have to check the first one
+	const auto ij = index[j];
+	const auto ik0 = index[k0];
+	const auto ik1 = index[k1];
+	if (!validity(ij) && !validity(ik0)) {
+		return true;
```
[the remark this answers is not in the document]
Comment (hawkfish) [reply src/function/aggregate/holistic/quantile.cpp:146]: It was a good thing I took another look here because the `NULL` handling was still not correct.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**